packages/terra-status-view/src/StatusView.jsx

```diff
@@ -36,223 +40,142 @@ const propTypes = {
-        const remainingHeight = (viewHeight > contentWithGlyphHeight) ? (viewHeight - contentWithGlyphHeight) : 0;
-        paddingTop = remainingHeight * 0.4;
-        paddingBottom = remainingHeight * 0.6;
-      }
-    } else if (!this.props.isAlignedTop) {
-      // glyph does not exist or does not fit, distribute remaining padding if any to center
-      const remainingHeight = viewHeight - contentHeight;
-      if (remainingHeight > 0) {
-        paddingTop = remainingHeight * 0.4;
-        paddingBottom = remainingHeight * 0.6;
-      }
-    }
-
-    // set which components can fit into the view
-    this.setState({
-      showGlyph,
-      paddingTop,
-      paddingBottom,
-    });
-  }
-
-  render() {
-    const {
-      children,
-      customGlyph,
-      isAlignedTop,
-      isGlyphHidden,
-      message,
-      title,
-      variant,
-      ...customProps
-    } = this.props;
-
-    const { intl } = this.context;
-
-    let glyphSection;
-    if (!isGlyphHidden && this.state.showGlyph) {
-      if (variant === StatusViewVariants.CUSTOM) {
-        glyphSection = (
-          <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}>
-            {customGlyph}
-          </div>
-        );
-      } else {
-        glyphSection = (
-          <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}>
-            <svg className={cx(variant)} />
-          </div>
-        );
-      }
-    }
+  return buttonAttrsArray.map(button => <Button {...button} className={cx(['button', button.className])} />);
+};
 
-    let messageSection;
-    if (message) {
-      messageSection = (
-        <div className={cx('message')} ref={(element) => { this.messageNode = element; }}>
-          {message}
+const StatusView = ({
+  buttonAttrs,
+  customGlyph,
+  isAlignedTop,
+  isGlyphHidden,
+  message,
+  title,
+  variant,
+  ...customProps
+}) => {
+  let glyphSection;
+  if (!isGlyphHidden) {
+    if (customGlyph) {
+      glyphSection = (
+        <div className={cx('glyph')}>
+          {customGlyph}
         </div>
       );
-    }
-
-    let actionSection;
-    if (React.Children.count(children) > 0) {
-      actionSection = (
-        <div className={cx('actions')} ref={(element) => { this.actionsNode = element; }}>
-          {children}
+    } else {
+      glyphSection = (
+        <div className={cx('glyph')}>
+          <svg className={cx(variant)} />
         </div>
       );
     }
+  }
 
-    let defaultTitle = title;
-    if (!defaultTitle) {
-      defaultTitle = (variant === StatusViewVariants.CUSTOM) ? '' : intl.formatMessage({ id: `Terra.status-view.${variant}` });
-    }
-
-    let dividerSection;
-    if (messageSection || actionSection) {
-      dividerSection = (
-        <div className={cx('divider')} ref={(element) => { this.dividerNode = element; }}>
-          <Divider />
-        </div>
-      );
-    }
+  let defaultTitle;
+  if (checkVariantExistence(variant)) {
```

Comment:
You should really only need to check that variant is defined, the propType defines which values are allowed, I don't think that we need to guard against improper usage of the API.

Reply:
if this check is removed, an erroneous title will render for a non existent variant - `Terra.status-view.{erroneous-title}` - should we let this happen versus not rendering anything? I'm fine either way

Reply:
I think we should. It's an invalid scenario either way and removing the check for a valid variant is less code to upkeep.

Reply:
removed